content/modules/flow-ui/pages/views/views.adoc

```diff
@@ -0,0 +1,402 @@
+
+[source,java,indent=0]
+----
+include::example$/onboarding/src/main/java/com/company/onboarding/view/user/UserListEventsView.java[tags=notifications-bean;after-close-event]
+----
+
+[[detach-event]]
+=== DetachEvent
+
+`DetachEvent` is fired before a view is detached from the UI. This event listener can be used for releasing resources acquired by the view.
+
+[source,java,indent=0]
+----
+include::example$/onboarding/src/main/java/com/company/onboarding/view/user/UserListEventsView.java[tags=detach-event]
+----
+
+NOTE: An event is fired by Vaadin, since a view is a UI component too.
+
+[[query-parameters-change-event]]
+=== QueryParametersChangeEvent
+
+`QueryParametersChangeEvent` is an event that informs which query parameters a view was opened with.
+
+[source,java,indent=0]
+----
+include::example$/onboarding/src/main/java/com/company/onboarding/view/user/UserListEventsView.java[tags=query-parameters-change-event]
+----
+
+[[init-entity-event]]
+=== InitEntityEvent
+
+`InitEntityEvent` is fired in views inherited from `StandardDetailView` before the new entity instance is set to edited entity container.
+
+Use this event listener to initialize default values in the new entity instance, for example:
+
+[source,java,indent=0]
+----
+include::example$/onboarding/src/main/java/com/company/onboarding/view/user/UserDetailEventsView.java[tags=init-entity-event]
+----
+
+[[validation-event]]
+=== ValidationEvent
+
+`ValidationEvent` is fired in views inherited from `StandardDetailView` when the view is validated on saving the view data context. Use this event listener to perform additional validation of the view.
+
+[source,java,indent=0]
+----
+include::example$/onboarding/src/main/java/com/company/onboarding/view/user/UserDetailEventsView.java[tags=validation-event]
+----
+
+[[before-save-vent]]
+=== BeforeSaveEvent
+
+`BeforeSaveEvent` is fired in views inherited from `StandardDetailView` before saving the view data context. Use this event listener to prevent saving, alter entity attribute values before saving or/and interact with the user before save.
+
+[source,java,indent=0]
+----
+include::example$/onboarding/src/main/java/com/company/onboarding/view/user/UserDetailEventsView.java[tags=before-save-event]
+----
+<1> - Interrupt the saving process if passwords do not match
+<2> - Set encoded password to the entity
+
+[[after-save-event]]
+=== AfterSaveEvent
+
+`AfterSaveEvent` is fired in views inherited from `StandardDetailView` after saving the view data context. Use this event listener to notify users after save, for example:
+
+[source,java,indent=0]
+----
+include::example$/onboarding/src/main/java/com/company/onboarding/view/user/UserDetailEventsView.java[tags=notifications-bean;after-save-event]
+----
+
+The diagram below shows the process of opening and closing a standard view:
```

Comment:
Let's move the diagrams to the beginning of the section, before the first event.